kroxylicious-api/src/main/java/io/kroxylicious/proxy/config/tls/SslProtocol.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.config.tls;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public enum SslProtocol {
```

Comment:
I'm not convinced that introducing this enum is the right thing to do.   Both the JDK and the Netty treat TLS Protocols name as a String.  I think we should do the same.   It is possible alternative TLS provider implementation name things in a different way.

Reply:
Use SSLContext.getDefault().getSupportedSSLParameters().getProtocols() to validate that the protocol names  in the allow and deny lists are known to the platform.   Log a warning if they are not. IOW do what you did for ciphersuites.